pkg/dynatrace-client/agent_version.go

```diff
@@ -0,0 +1,81 @@
+package dynatrace_client
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+func (dc *dynatraceClient) GetAgentVersionForIP(ip string) (string, error) {
+	if len(ip) == 0 {
```

Comment:
Some places we check for empty string by `len(str) == 0` and some places `str == ""`